src/main/scala/junctions/asyncqueue.scala

```diff
@@ -2,22 +2,28 @@
 
 package junctions
 import Chisel._
+import uncore.util.{AsyncResetRegVec, AsyncResetReg}
 
 object GrayCounter {
   def apply(bits: Int, increment: Bool = Bool(true)): UInt = {
-    val binary = RegInit(UInt(0, width = bits))
-    val incremented = binary + increment.asUInt()
-    binary := incremented
+    val incremented = Wire(UInt(width=bits))
+    val binary = AsyncResetReg(incremented, 0, Bool(true))
```

Comment:
I like to be clear about the reset value. But you're right about the Bool(true). I removed that.